### PR TITLE
added support to enable opt-in for forwarding exit spans without an entry span

### PR DESCRIPTION
**Pull request by Angith**
Changes

* Introduce an environment variable INSTANA_ALLOW_EXIT_AS_ROOT. Based on the environment variable, either the exit spans will be sent to the agent or suppressed. 
* Add unit tests for the above changes.

### CODE REVIEW

**Review by sanojsubran — CHANGES_REQUESTED** [review]
Please check the review comments

File: span.go [hunk 1]
```diff
@@ -87,6 +88,18 @@ func (r *spanS) FinishWithOptions(opts ot.FinishOptions) {
...
+	if r.context.Suppressed {
+		return false
+	} else if isExit && allowExitAsRoot {
+		return true
+	} else if isExit {
+		return false
+	}
+	return true
+}
+
```
[the remark this answers is not in the document]
Comment (Angith) [reply span.go]: It looks better, indeed 👍

File: span_test.go [hunk 2]
```diff
@@ -282,3 +283,133 @@ func TestSpan_Suppressed_SetTag(t *testing.T) {
...
+		opts          ot.StartSpanOptions
+	}
+	tests := []struct {
+		name      string
+		exportEnv bool
+		args      args
+		want      int
+	}{
+		{
+			name:      "env_disable_tag_false_exit_true",
```
[the remark this answers is not in the document]
Comment (Angith) [reply span_test.go]: Good suggestion. 👍

File: span_test.go [hunk 3]
```diff
@@ -282,3 +283,133 @@ func TestSpan_Suppressed_SetTag(t *testing.T) {
...
+				os.Unsetenv(allowExitAsRoot)
+			}
+
+			recorder := instana.NewTestRecorder()
+			tracer := instana.NewTracerWithEverything(&instana.Options{AgentClient: alwaysReadyClient{}}, recorder)
+			sp := tracer.StartSpanWithOptions(tt.args.operationName, tt.args.opts)
+			sp.Finish()
+			assert.Equal(t, tt.want, len(recorder.GetQueuedSpans()))
+		})
+	}
```
[the remark this answers is not in the document]
Comment (Angith) [reply span_test.go]: Yes. It looks good 👍

File: util.go [hunk 4]
```diff
@@ -18,6 +18,11 @@ import (
 	"time"
 
 	ot "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+const (
+	allowExitAsRoot = "INSTANA_ALLOW_EXIT_AS_ROOT"
```
[the remark this answers is not in the document]
Comment (Angith) [reply util.go]: done 👍

**Review by sanojsubran — APPROVED** [review]
Looks good to me. Good work @Angith !

**Pull request merged**